component/vdr/orb/vdr.go

```diff
@@ -632,14 +628,29 @@ func getRemovedSvcKeysID(currentService, updatedService []docdid.Service) []upda
 	return updateOpt
 }
 
-func getRemovedPKKeysID(currentVM, updatedVM []docdid.VerificationMethod) []update.Option {
+func getUpdatedPKKeysID(currentVM []docdid.VerificationMethod, didDoc *docdid.Doc,
+	sidetreePK map[string]*doc.PublicKey) []update.Option {
 	var updateOpt []update.Option
 
+	updatedVM := make([]docdid.Verification, 0)
+
+	updatedVM = append(updatedVM, didDoc.Authentication...)
+	updatedVM = append(updatedVM, didDoc.AssertionMethod...)
+	updatedVM = append(updatedVM, didDoc.CapabilityDelegation...)
+	updatedVM = append(updatedVM, didDoc.CapabilityInvocation...)
+	updatedVM = append(updatedVM, didDoc.KeyAgreement...)
+
+	existKeys := make(map[string]struct{})
+
 	for _, curr := range currentVM {
 		exist := false
 
 		for _, updated := range updatedVM {
-			if curr.ID == updated.ID {
+			if strings.Contains(curr.ID, updated.VerificationMethod.ID) {
+				if bytes.Equal(curr.Value, updated.VerificationMethod.Value) {
```

Comment:
Do we need to worry about VerificationMethod Type?

Reply:
yes

Reply:
it's fixed now